### PR TITLE
fixing another corner case for advanced indexing

### PR DESCRIPTION
**Pull request by ipdemes**
Fixing another corner case of advanced indexing when output from advanced indexing is 0-sized

### CODE REVIEW

File: tests/integration/test_advanced_indexing.py
Comment (ipdemes) [inline tests/integration/test_advanced_indexing.py:184]: this test doesn't work anymore because PointN type can't work when creating np.array(0). PointN type will be translated to one of the np.dtypes in the https://github.com/nv-legate/cunumeric/pull/473 so this test will work again

Comment (manopapad) [reply tests/integration/test_advanced_indexing.py:184]: I don't think the test will work still, because `fill` (namely the one you do on line 587 above) works only for fully-supported dtypes (e.g. np.float32 and np.int32).

Comment (ipdemes) [reply tests/integration/test_advanced_indexing.py]: That is right. Actually we don't need to call `fill` for `set_item`, because `out` will not get used if it's size is 0 in this case

**Review by manopapad — APPROVED** [review]
LGTM, feel free to merge once #473 is merged

**Comment by ipdemes** [comment]
I don't actually need #473 fir this PR anymore.

**Pull request merged**
